src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorHtmlDocument.cs

```diff
@@ -4,17 +4,22 @@
 #nullable disable
 
 using System;
+using System.Collections.Generic;
 
 namespace Microsoft.AspNetCore.Razor.Language;
 
 internal class DefaultRazorHtmlDocument : RazorHtmlDocument
 {
     private readonly string _generatedHtml;
     private readonly RazorCodeGenerationOptions _options;
+    private readonly SourceMapping[] _sourceMappings;
+    private readonly RazorCodeDocument _codeDocument;
 
     public DefaultRazorHtmlDocument(
+        RazorCodeDocument codeDocument,
         string generatedHtml,
-        RazorCodeGenerationOptions options)
+        RazorCodeGenerationOptions options,
+        SourceMapping[] sourceMappings)
```

Comment:
>SourceMapping[] [](http://example.com/codeflow?start=8&length=15)

I guess we cannot use `ImmutableArray` here?

Reply:
Strictly speaking here we probably could, but I tried to maintain consistency with RazorCSharpDocument. I'm only a visitor in the compiler and I didn't want to start rearranging all of the furniture :)